Sprint 11

AntonDemidenk0 opened the pull request.

AntonDemidenk0 committed:
Слил изменения из 10 спринта с тем что успел сделать в 11

AntonDemidenk0 committed:
Выполнено задание по 11 спринту

andrew-kuzn-yndx commented on Auth/AuthViewController.swift:
```diff
@@ -35,69 +35,106 @@ final class AuthViewController: UIViewController {
                 }
             case .failure(let error):
-                self.dismiss(animated: true)
+                UIBlockingProgressHUD.dismiss()
```
Дублирование UIBlockingProgressHUD. Его можно вывести за пределы switch, так как его вызывать в любом случае)

andrew-kuzn-yndx commented on Auth/SplashScreenVC.swift:
```diff
@@ -7,36 +7,95 @@ final class SplashViewController: UIViewController {
         }
         let tabBarController = UIStoryboard(name: "Main", bundle: .main)
-            .instantiateViewController(withIdentifier: "TabBarViewController")
+            .instantiateViewController(withIdentifier: "TabBarViewController") as! UITabBarController
```
А тут есть force cast( давай иначе это сделаем

andrew-kuzn-yndx commented on ProfileVC/ProfileViewController.swift:
```diff
@@ -42,26 +61,44 @@ final class ProfileViewController: UIViewController {
     
+    private func setupObserver() {
+        profileImageServiceObserver = NotificationCenter.default
+            .addObserver(forName: ProfileImageService.didChangeNotification, object: nil, queue: .main) { [weak self] _ in
```
Сейчас не принципиально, но в будущем всегда учитывай, что при добавлении Observer, нужно предусматривать и его удаление, чтобы не создать утечку памяти. 
Обычно это делается в один из заключительных этапов цикла жизни контроллера

AntonDemidenk0 committed:
Исправлены замечания:
- Убран  force cast в SplashScreenVC.swift
- Добавлен метод deinit для удаления observer
- Исправлен баг "Дебаггер печатает ошибку, а страница профиля показывает моковые данные" в случае, если в профиле юзера не заполнены все запрашиваемые поля, теперь подгружаются имеющиеся данные, остальные строки остаются пустыми
- Добавлен функционал кнопки логаута
